confluent_kafka/src/Producer.c

```diff
@@ -463,6 +479,17 @@ static PyMethodDef Producer_methods[] = {
 	  "callbacks may be triggered.\n"
 	  "\n"
 	},
+	{ "purge", (PyCFunction)Producer_purge, METH_VARARGS|METH_KEYWORDS,
+          ".. py:function:: purge([blocking])\n"
+          "\n"
+	  "   Purge messages in internal queues currently handled by producer instance.\n"
+	  "   The application will need to call poll() or flush() "
+	  "afterwards to serve the delivery report callbacks of the purged messages."
+	  "\n"
+          "  :param: bool blocking: If set to False, will not wait on background thread queue)\n"
```

Comment:
I think we should expose both  RD_KAFKA_PURGE_F_QUEUE  and RD_KAFKA_PURGE_F_INFLIGHT as an enum. In addition to the two options provided by librdkafka lest add RD_KAFKA_PURGE_F_ALL for both in-flight and queued messages. blocking should remain a bool though that's nicer than making users execute bitwise operators. 

See `AdminTypes.c` and `admin/__init__.py` for an example for exposing enums.